File: backend-node-typescript/src/features/continent/continent.controller.ts
import { Request, Response } from 'express';
import { ContinentService } from './continent.service';
import { ContinentRepository } from './continent.repository';

export class ContinentController {
  constructor(private continentService: ContinentService) { }

  async getItems(req: Request, res: Response): Promise<void> {
    const continents = await this.continentService.getItems();
    res.json(continents);
  }

}

const continentRepository = new ContinentRepository();
const continentService = new ContinentService(continentRepository);
export const continentController = new ContinentController(continentService);

File: backend-java/src/main/java/com/ganatan/backend_java/modules/continent/ContinentController.java
package com.ganatan.backend_java.modules.continent;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.List;

@RestController
public class ContinentController {
    
    private final ContinentService service;
    
    public ContinentController(ContinentService service) {
        System.out.println("[ganatan] ContinentController Constructor:");
        this.service = service;
    }
    
    @GetMapping("/continents")
    public List<Continent> getItems() {
        System.out.println("[ganatan] ContinentController getItems:");
        return service.getItems();
    }
}
